medmodels/treatment_effect_estimation/analysis_modules/estimate.py

```diff
@@ -0,0 +1,407 @@
+
+        Preconditions:
+            - Subject counts in each group must be non-zero to ensure valid
+                calculations.
+
+        Args:
+            medrecord (MedRecord): The MedRecord object containing the data.
+
+        Returns:
+            float: The calculated odds ratio between the treatment and control groups.
+
+
+        Raises:
+            AssertionError: If preconditions are not met, indicating potential issues
+                with group formation or subject count retrieval.
+        """
+        (
+            num_treat_true,
+            num_treat_false,
+            num_control_true,
+            num_control_false,
+        ) = self._compute_subject_counts(medrecord=medrecord)
+
+        return (num_treat_true / num_control_true) / (
+            num_treat_false / num_control_false
+        )
+
+    def confounding_bias(self, medrecord: MedRecord) -> float:
+        """
+        Calculates the confounding bias (CB) to assess the impact of potential
+        confounders on the observed association between treatment and outcome. A
+        confounder is a variable that influences both the dependent (outcome) and
+        independent (treatment) variables, potentially biasing the study results.
+
+        Interpretation of CB:
+        - CB = 1 indicates no confounding bias.
+        - CB != 1 suggests the presence of confounding bias, indicating potential
+            confounders.
+
+        The method relies on the relative risk (RR) as an intermediary measure and
+        adjusts the observed association for potential confounding effects. This
+        adjustment helps in identifying whether the observed association might be
+        influenced by factors other than the treatment.
+
+        Precondition:
+            - Subject counts in each group must be non-zero to avoid division by zero
+                errors.
+
+        Args:
+            medrecord (MedRecord): The MedRecord object containing the data.
+
+        Returns:
+            float: The calculated confounding bias.
+        """
+        (
+            num_treat_true,
+            num_treat_false,
+            num_control_true,
+            num_control_false,
+        ) = self._compute_subject_counts(medrecord)
+        relative_risk = self.relative_risk(medrecord)
+
+        if relative_risk == 1:
+            return 1
+
+        multiplier = relative_risk - 1
+        numerator = (
+            num_treat_true / (num_treat_true + num_treat_false)
+        ) * multiplier + 1
+        denominator = (
+            num_control_true / (num_control_true + num_control_false)
+        ) * multiplier + 1
+
+        return numerator / denominator
+
+    def absolute_risk(self, medrecord: MedRecord) -> float:
+        """
+        Calculates the absolute risk (AR) of an event occurring in the treatment group
+        compared to the control group. AR is a measure of the incidence of an event in
+        each group.
+
+
+        Preconditions:
```

Comment:
I don't think preconditions are part of the google python docstring styleguide. This should probably just be handled in the Raises part

Reply:
also applies to some other functions in this file